pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -691,11 +693,19 @@ func (c *Controller) setupIndex() *AmbientIndex {
 	return &idx
 }
 
-func (a *AmbientIndex) handlePod(_, newObj any, isDelete bool, c *Controller) sets.Set[model.ConfigKey] {
+func (a *AmbientIndex) handlePod(oldObj, newObj any, isDelete bool, c *Controller) sets.Set[model.ConfigKey] {
+	p := controllers.Extract[*v1.Pod](newObj)
+	old := controllers.Extract[*v1.Pod](oldObj)
+	if old != nil {
+		// compare only labels and pod phase, which are what we care about
+		if reflect.DeepEqual(old.Labels, p.Labels) && old.Status.Phase == p.Status.Phase &&
```

Comment:
maps.Equal is preferred, faster than reflect

Reply:
pod IP is allocated before pod running, before that pod status won't be ready and phase not be running

Reply:
So that means after ip allocated the phase must change